### PR TITLE
Add workflow for uploading baseline images as a release asset

### PR DESCRIPTION
**Pull request by meghanrjones**
**Description of proposed changes**

Based on https://github.com/GenericMappingTools/pygmt/pull/1317

Closes https://github.com/GenericMappingTools/gmt/issues/5724

**Reminders**

- [ ] Make sure that your code follows our style. Use the other functions/files as a basis.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Describe changes to function behavior and arguments in a comment below the function declaration.
- [ ] If adding new functionality, add a detailed description to the documentation and/or an example.

### CODE REVIEW

File: .github/workflows/release-baseline-images.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    - name: Checkout
+      uses: actions/checkout@v3.0.2
+
+    - name: Setup data version control (DVC)
+      uses: iterative/setup-dvc@v1.0.3
+
+    - name: Pull baseline image data from dvc remote
+      run: |
+        dvc pull
+        ls -lhR test/baseline/
```
Comment (seisman) [inline .github/workflows/release-baseline-images.yml:27]: I think we also need the images in `doc/examples` and `doc/scripts`

Comment (meghanrjones) [reply .github/workflows/release-baseline-images.yml:27]: These are included in the release tarball after https://github.com/GenericMappingTools/gmt/pull/6533, but I added them to the baseline-images asset as well in https://github.com/GenericMappingTools/gmt/pull/6782/commits/aa93c0693d8decf6ae0b41fd3a63663e7ec12d69. This change increases the size from ~40 MB to ~70 MB.

**Comment by meghanrjones** [comment]
@seisman, do you think having this as a separate workflow is better than zipping up the files as part of the [build-release script](https://github.com/GenericMappingTools/gmt/blob/master/admin/build-release.sh) and adding the upload step to the [draft-release](https://github.com/GenericMappingTools/gmt/blob/master/.github/workflows/draft-release.yml) workflow? I am undecided about which would be preferable.

**Comment by meghanrjones** [comment]
Right, it would need to be run by Paul and placed on the GMT FTP server with the other files. @PaulWessel, any opinions on whether to bundle and upload the baseline images manually as part of `build-release.sh` along with the other files or keep it separate as currently written in this PR?

**Comment by PaulWessel** [comment]
I think it is probably fine as you have it in the CI - maybe one day the rest will follow.

**Comment by meghanrjones** [comment]
> In terms of release timing: I have all Saturday and maybe a short slice on Sunday but then it is in the car with the Watts and off to some serious hiking in the Lake District. Probably not going to be productive at some mountain cabin. Then on Tuesday I will be dropped off in Newcastle and take the train to Edinburgh for a Uni visit and talk. Then a 6 hour train trip back on Wednesday. So let's see if we can get lots of stuff done before then - otherwise the release may be end of next week.

Sounds fun!
I can update the changelog, citation info, and test with PyGMT tomorrow AM EST. I think this is ready if I can get an approval on it.

**Review by seisman — APPROVED** [review]

**Pull request merged**